pkg/cloud-provider/cloudapi/azure/azure_security.go

```diff
@@ -766,11 +768,24 @@ func (c *azureCloud) CreateSecurityGroup(addressGroupIdentifier *securitygroup.C
 	return to.StringPtr(cloudSecurityGroupID), nil
 }
 
+// UpdateSecurityGroupRules invokes cloud api and performs rule update on the cloud using security group's full set of rules in targetRules.
 func (c *azureCloud) UpdateSecurityGroupRules(addressGroupIdentifier *securitygroup.CloudResource,
-	ingressRules []*securitygroup.IngressRule, egressRules []*securitygroup.EgressRule) error {
+	_, _, targetRules []*securitygroup.CloudRule) error {
```

Comment:
change it to all rules.. comment would be invokes cloud api and update cloud security group with allRules.